float64[6] kd     # vd = kd*f
float64[6] fd
float64 timeout     # maximum moving time
float64 z_thresh  # stop when z < z_thresh
